sync/class.jetpack-sync-actions.php

```diff
@@ -227,61 +227,43 @@ static function jetpack_cron_schedule( $schedules ) {
 		return $schedules;
 	}
 
-	// try to send actions until we run out of things to send,
-	// or have to wait more than 15s before sending again,
-	// or we hit a lock or some other sending issue
 	static function do_cron_sync() {
-		if ( ! self::sync_allowed() ) {
-			return;
-		}
-
-		self::initialize_sender();
-
-		$time_limit = Jetpack_Sync_Settings::get_setting( 'cron_sync_time_limit' );
-		$start_time = time();
-
-		do {
-			$next_sync_time = self::$sender->get_next_sync_time( 'sync' );
-
-			if ( $next_sync_time ) {
-				$delay = $next_sync_time - time() + 1;
-				if ( $delay > 15 ) {
-					break;
-				} elseif ( $delay > 0 ) {
-					sleep( $delay );
-				}
-			}
-
-			$result = self::$sender->do_sync();
-		} while ( $result && ( $start_time + $time_limit ) > time() );
+        self::do_cron_sync_by_type( 'sync' );
 	}
 
 	static function do_cron_full_sync() {
-		if ( ! self::sync_allowed() ) {
-			return;
-		}
-
-		self::initialize_sender();
-
-		$time_limit = Jetpack_Sync_Settings::get_setting( 'cron_sync_time_limit' );
-		$start_time = time();
-
-		do {
-			$next_sync_time = self::$sender->get_next_sync_time( 'full_sync' );
-
-			if ( $next_sync_time ) {
-				$delay = $next_sync_time - time() + 1;
-				if ( $delay > 15 ) {
-					break;
-				} elseif ( $delay > 0 ) {
-					sleep( $delay );
-				}
-			}
-
-			$result = self::$sender->do_full_sync();
-		} while ( $result && ( $start_time + $time_limit ) > time() );
+        self::do_cron_sync_by_type( 'full_sync' );
 	}
 
+    // try to send actions until we run out of things to send,
+    // or have to wait more than 15s before sending again,
+    // or we hit a lock or some other sending issue
+    static function do_cron_sync_by_type( $type ) {
```

Comment:
Looks like you had a few more spaces left. I took care of them in 60e54701d6a12dbeae5c1512fd319926493f6591

I also updated the function doc comment to avoid triggering phpcs warnings.